### PR TITLE
add package for flux-pmix

### PR DESCRIPTION
**Pull request by grondo**
This PR adds a package for flux-pmix, a job shell plugin for Flux which allows boostrapping OpenMPI v5.0+ executables.

Edit: requisite version of pmix 4.1.0 is also added as part of this pr

### CODE REVIEW

**Comment by rhc54** [comment]
Hi Mark! I'm afraid I am ignorant of Spack, so perhaps someone else should review this. I will take a look at the flux-pmix repo, however, just to ensure that all the info required by OMPI and others is present.

**Comment by grondo** [comment]
@rhc54, that would be great! I think spack required your review on this PR because we had to add pmix v4.1.0 for the flux-pmix implementation to build against. If you can just ACK that change, I think a spack maintainer can review the rest of the PR. Thanks!

**Review by rhc54 — DISMISSED** [review]
Adding in PMIx v4.1.0 is great - thanks for doing it!

**Comment by grondo** [comment]
Thanks @rhc54!

Also, if you find problems or have any comments on the upstream flux-pmix repo, feel free to open an issue/issues!
Thanks!

File: var/spack/repos/builtin/packages/flux-pmix/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/flux-pmix/package.py]: In general it is best to state the real requirements to build a software in `package.py`, regardless of what you might find on most systems. Then if there's a need to cut build-time you can:
```console
spack external find <pkg>
``` 
to use what is installed already outside of Spack. Are there other build dependencies that are needed and not listed?

File: var/spack/repos/builtin/packages/flux-pmix/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/flux-pmix/package.py:36]: This suggests we might be missing a:
```
depends_on('autoconf', type='build')
depends_on('automake', type='build')
...
```
Can you double check that?

**Review by alalazo — CHANGES_REQUESTED** [review]
A question on build dependencies, but otherwise LGTM.

File: var/spack/repos/builtin/packages/flux-pmix/package.py
Comment (grondo) [reply var/spack/repos/builtin/packages/flux-pmix/package.py]: I don't think there are any extra runtime dependencies over what is listed in the `package.py` 

```console
 grondo@quartz1538:~/git/spack$ spack external find flux-pmix
==> No new external packages detected
```
Thank you for the hint about the useful command.

File: var/spack/repos/builtin/packages/flux-pmix/package.py
Comment (grondo) [reply var/spack/repos/builtin/packages/flux-pmix/package.py:36]: Yes, `flux-pmix` requires `autoconf`, `automake` and `libtool` at build time. I will add those. Thanks!

Comment (grondo) [reply var/spack/repos/builtin/packages/flux-pmix/package.py:36]: I've added these packages as build dependencies and pushed the result.

File: var/spack/repos/builtin/packages/flux-pmix/package.py
Comment (grondo) [reply var/spack/repos/builtin/packages/flux-pmix/package.py]: Actually, looking again, the package does explicitly require `libczmq` and `jansson`, but these are transitively  satisfied by the `flux-core` dependency. Is this ok, or do you prefer these be added to the `package.py`?

Comment (alalazo) [reply var/spack/repos/builtin/packages/flux-pmix/package.py]: What I meant is that `pkg-config` and maybe `auto*` packages should be listed as build dependencies if they are needed. Then you can:
```console
% spack external find pkg-config
```
if you want to reuse `pkg-config` from your host instead of having Spack build it. I was pointing this out since the impression I got from the comment was that `pkg-config` was added only because some system ship a version that is too old, while my advice is to always add it if it is a dependency.

Comment (grondo) [reply var/spack/repos/builtin/packages/flux-pmix/package.py]: Ah, I understand now. That makes sense, thanks. 

I will remove the confusing comment then. I have already added the autotools as a build dependency.

Apologies, I'm a spack newbie!

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/flux-pmix/package.py
Comment (alalazo) [reply var/spack/repos/builtin/packages/flux-pmix/package.py]: No worries, thanks @grondo !

**Comment by grondo** [comment]
Thank you!

**Pull request merged**
